source/agora/consensus/data/Enrollment.d

```diff
@@ -68,6 +68,14 @@ public struct Enrollment
     }
 }
 
+/// Results after adding the validator's enrollment data
+public enum ValidatorAddResult : ubyte
+{
+    Failure,    //  if an error occurs,
+    Invalid,    //  if the enrollment is abnormal,
+    Success     //  if the enrollment is added normally
+}
+
```

Comment:
That's not a way we want to go. Having `Failure` and `Invalid` is quite confusing for the caller, and probably won't be handled correctly. It is also a [leaky abstraction](https://www.joelonsoftware.com/2002/11/11/the-law-of-leaky-abstractions/).

Reply:
I think so too, I should change it again in a first way. it was complicated to modify the unit test.